### PR TITLE
Add pause and unpause missing validations

### LINKED ISSUE #7980: Add validation for pause key in PauseLogic and UnpauseLogic
### Problem

After investigation of #7822
It seem that there are some missing validations in some of the logics that were copied from services.

PauseLogic, UnpauseLogic
Missing logic -> validateTrue(hasPauseKey(), TOKEN_HAS_NO_PAUSE_KEY);

### Solution

Add the pause key validation logics and fix failing tests

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by zhpetkov**
**Description**:
After changes, we don't have missing logic that validates the pause state of the token.  I add `changePauseStatus` in `Token` which modifies the state of the `pause` property.  Before the property modification, the method performs validation, that the respective token has a pause key. I replace `setPause` with `changePauseStatus` in `PauseLogic`, and `UnpauseLogic` because this will help to properly change the state of the token. Make needed modifications in tests. The changes are borrowed from Hedera Services. 

**Related issue(s)**:

Fixes #7980

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by steven-sheehy — COMMENTED** [review]
Did we verify acceptance tests with this change?

**Comment by zhpetkov** [comment]
> Did we verify acceptance tests with this change?

Yes, they passed successfully.

File: hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/PauseLogic.java [hunk 1]
```diff
@@ -41,7 +41,7 @@ public void pause(final Id targetTokenId, final Store store) {
         var token = store.loadPossiblyPausedToken(targetTokenId.asEvmAddress());
 
         /* --- Do the business logic --- */
-        var pausedToken = token.setPaused(true);
+        var pausedToken = token.changePauseStatus(true);
```
Comment (0xivanov) [inline hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/PauseLogic.java:44]: remove `setPaused` method, and replace it with `changePauseStatus` in `HederaTokenStoreTest`

File: hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/FreezeLogic.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
  * <p>
  * Differences with the original:
  * 1. Use abstraction for the state by introducing {@link Store} interface
- * 2. Used tokenRelationship.setFrozen instead of tokenRelationship.changeFrozenState (like in services)
+ * 2. Used tokenRelationship.changeFrozenState (like in services)
```
Comment (0xivanov) [inline hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/FreezeLogic.java:33]: `changeFrozenState` is not present in this PR, is it supposed to be in another PR?

Comment (zhpetkov) [reply hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/FreezeLogic.java:33]: Yes, but PR for freeze and unfreeze is merged. The changes associated with freeze logic are only in this comment. I can remove them.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by kselveliev — APPROVED** [review]
LGTM
